File: examples/python/spi.py
#!/usr/bin/env python

# Author: Henry Bruce <henry.bruce@intel.com>
# Copyright (c) 2016 Intel Corporation.
#
# SPDX-License-Identifier: MIT

# Example Usage: Read from MCP3004 ADC pin 0 in single ended mode

import mraa
import time

# initialise SPI
dev = mraa.Spi(0)

# prepare data to send
txbuf = bytearray(3)
txbuf[0] = 0x01
txbuf[1] = 0x80
txbuf[2] = 0x00

while True:
    # send data through SPI
    rxbuf = dev.write(txbuf)
    value = ((rxbuf[1] & 0x03) << 8) | rxbuf[2]
    print(value)

    time.sleep(0.5)

File: examples/javascript/spi.js
#!/usr/bin/env node

/*
 * Author: Brendan Le Foll <brendan.le.foll@intel.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * SPDX-License-Identifier: MIT
 */

"use strict";

const mraa = require('mraa'); //require mraa

let spiDevice = new mraa.Spi(0);
let buf = new Buffer(4);
buf[0] = 0xf4;
buf[1] = 0x2e;
buf[2] = 0x3e;
buf[3] = 0x4e;
let buf2 = spiDevice.write(buf);
console.log("Sent: " + buf.toString('hex') + ". Received: " + buf2.toString('hex'));
